### DISCUSSION TITLE
[IKE] unable to resolve 1.1.1.138, initiate aborted

### DISCUSSION THREAD

**Discussion by jsaak** [strongswan/strongswan#2502] [Q&A] [answered]
I am getting this error, when initiating a tunnel
```
$ swanctl -i -i gw-gw-vodafone
[IKE] unable to resolve 1.1.1.138, initiate aborted
```
https://github.com/strongswan/strongswan/blob/5.9.8/src/libcharon/sa/ike_sa.c#L1549

Any idea why i get this? I am guessing ipv6-ipv4 address family mixup. No idea really, the code is a bit complicated for me to understand. Is there a way to force the config to only use ipv4?

(ip-s changed to 1.1.1.x)

swanctl.conf:
```
connections {
  gw-gw-vodafone {
      local_addrs  = 1.1.1.59
      remote_addrs =  1.1.1.138

      local {
         auth = psk
         id = 1.1.1.59
      }

      remote {
         auth = psk
         id = 1.1.1.138
      }

      children {
         net-net-pe {
             local_ts  = 0.0.0.0/0
             remote_ts = 10.0.1.0/24
             rekey_time = 3600
             esp_proposals = aes256-sha256-modp4096, aes256-sha256, aes256-sha256-modp2048, default
             start_action=start
         }
      }

      version = 2
      mobike = no
      reauth_time = 36000
      proposals = aes256-sha256-modp4096 ,default
     }
} 
```

**Comment by tobiasbrunner** [comment]
One possibility is that there is no socket open on IPv4, which disables IPv4 as address family. Check the log for errors during startup.

**Comment by jsaak** [comment]
Thank you for your comment! It steered me to the right direction. Charon was running twice. After stopping the other charon (which used an empty ipsec.conf), everything worked as expected (using swanctl.conf). I did not found any error message indicating that this was a case. The only error i got ("unable to resolve <valid ipv4 address>") is clearly non-sense. At least a message that ipv4 is disabled would help.

Debian packaging is highly confusing for me. strongswan depends on strongswan-charon and strongswan-starter. It seems that i do not need any of it? Instead i need charon-systemd, strongswan-swanctl, strongswan-libcharon???

Also the dependencies are broken, i think, in debian-stable
I saw that, for example 
strongswan-libcharon
Breaks: strongswan-starter (<= 5.6.1-2)
Replaces: strongswan-starter (<= 5.6.1-2)

strongswan-swanctl
Breaks: strongswan-starter (<< 5.8.0-1)
Replaces: strongswan-starter (<< 5.8.0-1)

But i had strongswan-starter 5.9 which it seems is legal according to apt, but should not be.

**Reply by tobiasbrunner** [reply]
> Debian packaging is highly confusing for me. strongswan depends on strongswan-charon and strongswan-starter. It seems that i do not need any of it? Instead i need charon-systemd, strongswan-swanctl, strongswan-libcharon???

See the note regarding Debian/Ubuntu packages in the [installation docs](https://docs.strongswan.org/docs/5.9/install/install.html), and e.g. the discussion in #2282 (there are others).


> Also the dependencies are broken, i think, in debian-stable

No idea. Maybe [report it directly to the Debian maintainers](https://bugs.debian.org/cgi-bin/pkgreport.cgi?repeatmerged=no&src=strongswan) if you think there is an issue.

**Reply by jsaak** [reply]
I did report it to debian maintainers.

I do not know how to improve the situation further. I wasted 2 days of my life.
I hope you added a meaningful error message after 5.9. If not, maybe it would be a good idea to do it now?

**Reply by tobiasbrunner** [reply, marked as answer]
Thanks for reporting it.

> I hope you added a meaningful error message after 5.9.

There should already be an error logged that sockets could not be opened (something like `unable to bind socket: Address already in use`, followed by `could not open IPv4 socket, IPv4 disabled`).

**Reply by jsaak** [reply]
And you are right. I could grep these errors from syslog. No idea why i did not see them before. Thank you!
